### PR TITLE
more correct continuous collision detection for thin objects

### PR DESCRIPTION
**Pull request by AndrewMeadows**
- fix to prevent thin objects from falling through thin shapes

### CODE REVIEW

File: libraries/physics/src/ObjectMotionState.cpp [hunk 1]
```diff
@@ -155,26 +155,25 @@ void ObjectMotionState::setMotionType(PhysicsMotionType motionType) {
...
+        transform.setIdentity();
+        _shape->getAabb(transform, aabbMin, aabbMax);
+        aabbMin = aabbMax - aabbMin;
+        btScalar radius = *((btScalar*)(aabbMin) + aabbMin.minAxis());
+
+        // use the minimum dimension as the radius of the CCD proxy sphere
+        _body->setCcdSweptSphereRadius(radius);
+
+        // also use the radius as the motion threshold for enabling CCD
+        _body->setCcdMotionThreshold(radius);
```
Comment (AndrewMeadows) [inline libraries/physics/src/ObjectMotionState.cpp]: This is the real fix right here: we're using a much smaller `CcdMotionThreshold` for thin objects.

File: libraries/physics/src/ObjectMotionState.cpp [hunk 2]
```diff
@@ -188,8 +187,8 @@ void ObjectMotionState::setRigidBody(btRigidBody* body) {
         if (_body) {
             _body->setUserPointer(this);
             assert(_body->getCollisionShape() == _shape);
+            updateCCDConfiguration();
         }
-        updateCCDConfiguration();
```
Comment (AndrewMeadows) [inline libraries/physics/src/ObjectMotionState.cpp]: The rest of this stuff was just to remove the reduntant check for valid `_body` in `ObjectMotionState::updateCCDConfiguration()`.

**Comment by AndrewMeadows** [comment]
**Test Plan**:

**(1)** Make a big static box as your **floor**.  Make sure its vertical dimension is a meter or more.  This thick convex object will prevent you from losing the dynamic "tunnelling" object used in this test.

**(2)** Make a thin static box of a different color with dimensions `{ x: 2.0, y: 0.05, z: 2.0 }`.  Place it about above the floor by half a meter or more.  This will be called the **obstacle**.

**(3)** Make a thin dynamic box of yet a third color with dimensions `{ x: 0.2, y: 0.01, z: 0.2 }`.  Make it dynamic and give it a gravity of `{ x: 0, y: -15, z: 0 }`.  This will be called the **tunneler**.

**(4)** Maneuver the **tunneler** to be on the **floor** under the **obstacle**.

**(5)** Pull the **tunneler** up to be several meters above the **obstacle** and let it fall.
--> on hifi/master the **tunneler** will most likely fall through the **obstacle** and hit the **floor**
--> on this PR the **tunneler** will hit the **obstacle** and not pass through

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11620/HighFidelity-Beta-PR13707-cf04a91.apk

**Comment by AndrewMeadows** [comment]
The internal manuscript case for this is 17085.
https://highfidelity.manuscript.com/f/cases/17085/thin-objects-fall-through-mesh-shapes

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11620/HighFidelity-Beta-PR13707-cf04a91.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11620/HighFidelity-Beta-Interface-PR13707-cf04a91.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11620/HighFidelity-Beta-PR13707-cf04a91.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11620/HighFidelity-Beta-Interface-PR13707-cf04a91.exe

**Comment by qliejahner** [comment]
Currently testing

**Pull request merged**